irestweaks.php

```diff
@@ -5,6 +5,44 @@
 use CRM_Irestweaks_ExtensionUtil as E;
 // phpcs:enable
 
+/**
+ * Implements hook_civicrm_postProcess().
+ *
+ * @link http://wiki.civicrm.org/confluence/display/CRMDOC/hook_civicrm_postProcess
+ */
+function irestweaks_civicrm_postProcess($formName, &$form) {
+  // Detect contribution page and contribution confirm page
+  if ($formName === 'CRM_Contribute_Form_Contribution_Confirm') {
+    // Get the source param by the entryURL in the controler array
+    $controller = $form->getVar('controller');
+    // Process to get the source param
+    $params = explode('?', $controller->_entryURL);
+    parse_str(end($params), $parseURL);
+
+    $getSource = '';
+
+    // Remove amp; since it was not remove using parse_str
+    foreach ($parseURL as $key => $value) {
+      $newKey = str_replace('amp;', '', $key);
+
+      if ($newKey === 'source') {
+        $getSource = 'Online Contribution: ' . $form->_values['title'] . ' - ' . $value;
+
+        $query = "
```

Comment:
To update a record in civicrm, we must use the api, instead of a direct sql query. In this case we need to use the `contribution.create` api, passing in the `id` parameter so the api knows which contribution to update.